backends-velox/src/main/scala/io/glutenproject/execution/ShuffledHashJoinExecTransformer.scala

```diff
@@ -172,24 +173,8 @@ case class GlutenBroadcastHashJoinExecTransformer(
     right,
     isNullAwareAntiJoin) {
 
-  override protected lazy val substraitJoinType: JoinRel.JoinType = joinType match {
-    case Inner =>
-      JoinRel.JoinType.JOIN_TYPE_INNER
-    case FullOuter =>
-      JoinRel.JoinType.JOIN_TYPE_OUTER
-    case LeftOuter | RightOuter =>
-      // The right side is required to be used for building hash table in Substrait plan.
-      // Therefore, for RightOuter Join, the left and right relations are exchanged and the
-      // join type is reverted.
-      JoinRel.JoinType.JOIN_TYPE_LEFT
-    case LeftSemi | ExistenceJoin(_) =>
-      JoinRel.JoinType.JOIN_TYPE_LEFT_SEMI
-    case LeftAnti =>
-      JoinRel.JoinType.JOIN_TYPE_ANTI
-    case _ =>
-      // TODO: Support cross join with Cross Rel
-      JoinRel.JoinType.UNRECOGNIZED
-  }
+  override protected lazy val substraitJoinType: JoinRel.JoinType =
+    SubstraitUtil.toSubstrait(joinType)
```

Comment:
We exchange tables sometimes to adjust the build side of suffled hash join for performance optimization. For BHJ and SMJ, we may need to keep the original left and right tables.

Reply:
Reverted this change. Instead changed `Inner` to `InnerLike` which will work for both `Inner` & `Cross` joins.